hw/ip/i2c/data/i2c_testplan.hjson

```diff
@@ -445,4 +445,144 @@
       tests: ["i2c_target_hrst"]
     }
   ]
+  covergroups: [
+    {
+      name: i2c_interrupts_cg
+      desc: '''
+            Cover all the interrupts raised during I2C operation in both host and target mode
+            Cross interrupt with INTERRUPT_TEST register
+            '''
+    }
+    {
+      name: i2c_fifo_reset_cg
+      desc: '''
+            Cover the FIFO reset bits in FIFO_CTL register
+            Cross fmt_threshold with FMTRST
+            Cross rx_threshold with RXRST
+            Cross fmt_overflow with FMTRST
+            Cross rx_overflow with RXRST
+            Cross acq_overflow with ACQRST
+            Cross tx_overflow with TXRST
+            '''
+    }
+    {
+      name: i2c_fifo_level_cg
+      desc: '''
+             Cover the trigger level for FMT_FIFO and RX_FIFO interupts
+             * Cross fmt_threshold interrupt with FMT_FIFO trigger level
+             * Cross rx_threshold interrupt with RX_FIFO trigger level
+             Cover the FIFO levels in FIFO_STATUS
+            '''
+    }
+    {
+      name: i2c_operating_mode_cg
+      desc: '''
+            * Cover the operating mode( Host/Target) of DUT
+            * Cover the operating mode( Host/Target) of TB
+            * Cross the operating modes of DUT and TB
+            * Cover the SCL frequency
+            * Cross the SCL frequency with operating mode of DUT
+            '''
+    }
+    {
+      name: i2c_rd_wr_cg
+      desc: '''
+            Cover the number of bytes read from Target
+            Cover the number of bytes written to Target
+            * Cross the number of bytes read in Target and Host mode of DUT
+            * Cross the number of bytes written in Target and Host mode of DUT
```

Comment:
Am I right to say that this would create a cross of `256*256`? We should probably find a smarter way of doing this. Maybe make a cross based on `clog2` of the number of Bytes?

Reply:
Not necessarily, we don't need to cover all the values of read/write bytes, instead we can select scalar range bins, for instance the following bins can be used

`read_byte : coverpoint <variable> {`
`    bins all_zero = 8'h00;`
`    bins all_ones = 8h'FF; `
`    bins small = [8'd1: 8'100];`
`    bins medium = [8'd101: 8'd200];`
`    bins high = [8'd201:8'd256];`
`}`